client/app.jsx

```diff
@@ -1,23 +1,49 @@
 import React from 'react';
-import Home from './pages/dt-home';
-import Mobile from './pages/mb-home';
 import parseRoute from './lib/parse-route';
+import RoomCreation from './components/dt-start';
+import DesktopGame from './components/dt-play-answer';
+import MobileGame from './components/mb-play-answer';
+import PlayerCreation from './components/mb-start';
 
 export default class App extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      route: parseRoute(window.location.hash)
+      route: parseRoute(window.location.hash),
+      gameId: null,
+      prompts: []
     };
+    this.updateGameId = this.updateGameId.bind(this);
+  }
+
+  componentDidMount() {
+    window.addEventListener('hashchange', () => {
+      this.setState(
+        { route: parseRoute(window.location.hash) }
+      );
+    });
+  }
+
+  updateGameId(result) {
+    this.setState({ gameId: result });
+  }
+
+  updatePrompts(result) {
+    this.setState({ prompts: result });
   }
 
   renderPage() {
-    const { path } = this.state.route;
-    if (path === '') {
-      return <Home />;
-    }
-    if (path === 'mobile') {
-      return <Mobile />;
+    const { route, gameId, prompts } = this.state;
+    if (route.path === '') {
+      return <RoomCreation
+        data={ { gameId: gameId, updateGameId: this.updateGameId.bind(this) } } />;
+    } else if (route.path === 'mobile') {
+      return <PlayerCreation
+        data={ { gameId: gameId, updateGameId: this.updateGameId.bind(this) } }/>;
+    } else if (route.path === 'game') {
+      return <DesktopGame data={ { gameId, updatePrompts: this.updatePrompts.bind(this) } }/>;
+    } else if (route.path === 'play') {
+      return <MobileGame data={ { gameId, prompts } }/>;
```

Comment:
This `data` prop thing might seem like a good idea, but it's an anti-pattern. It's much easier to understand the component code for MobileGame, PlayerCreation, etc, if each prop is given its own name. 

_Sometimes_ it makes sense to pass an object as a prop, but that's usually when the object is just "dumb" data (a data model with no methods).